### ISSUE TITLE
Private Docker repo

### ISSUE THREAD

**Issue by bobbyhubbard** [pivotal/kpack#296]
Forgive me if this question has already been asked. I'm attempting to install kpack in an internal kube cluster that is locked to local images only. So far I have pulled, scanned/inspected and then signed the kpack images and have been able to get most of the system functional. My problem comes when I add my image and the controller emits an error. I'm assuming this is trying to pull another image? Maybe the buildpack? It doesnt tell me what its actually trying to grab. 

> unable to fetch remote run image: Get https://index.docker.io/v2/: dial tcp 34.193.164.221:443: connect: connection refused

So question is... A) What is it and B) Can I configure it to pull from my corporate docker repo?

**Comment by ashwin-venkatesh** [comment]
hey @bobbyhubbard 

I think I have an understanding of the problem you are running into. But just to be sure, you were able to relocate all of the kpack images into your private cluster. I guess you relocated the builder image as well. The builder image references a run image in it's metadata which is an image on a public dockerhub. 

Addressing #190 should fix this problem but has not been done yet.

In it's current format, to run kpack entirely in a private env, you'll need to use the experimental resources of the store and the stack. You can populate the store with a builder to get started and relocate the run image and a build image locally and then create a  customer builder by referencing them. This is a non-trivial amount of work unfortunately.

**Comment by ashwin-venkatesh** [comment]
To add more context, a run image is the image that the final is is built on. The builder image has a reference to it internally. The builder also supports the concept of run image mirrors, which are copies of the same run image but on other registries. These images are public images which is why it works fine on most envs.
In an "air-gapped" situation, kpack cannot access the run image which leads to the builds to fail. A way of addressing this is by being able to pass in additional run image mirrors to the builder config, which is what #190 aims to deliver which allows kpack users to provide additional mirrors which can work in a situation similar to yours.

But kpack currently does support experimental resources which allow the user to now move all of the building blocks into their cluster and create their own builders. This is a little more complicated but offers a much higher degree of flexibility. This allows you to explicitly provide every image that will be required for the lifecycle of an image build and can hence work in environments similar to yours.

**Comment by mgibson1121** [comment]
@bobbyhubbard Thanks for checking in.  Two questions, were @ashwin-venkatesh's comments helpful in resolving your issue?  Second, would you be willing to talk to us a little bit more about your use-case and your experience using kpack so far in the context of a user interview?

**Comment by matthewmcnew** [comment]
Closing due to due to lack of activity.

**Comment by cmoulliard** [comment]
I'm faced to the same issue as reported here which is "How can we configure kpack to access a docker registry deployed on the same k8s cluster as where kpack is running ?"

```
kapp: Error: waiting on reconcile builder/cf-default-builder (kpack.io/v1alpha1) namespace: cf-workloads-staging:
  Finished unsuccessfully (Encountered failure condition Ready == False:  (message: Get "https://kind-registry.infra.svc:5000/v2/": dial tcp: lookup kind-registry.infra.svc on 10.96.0.10:53: no such host))
```
The docker registry service exists under the namespace `infra` and is running on port 5000 
```
kc get svc -n infra
NAME            TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
kube-registry   ClusterIP   10.97.220.108   <none>        5000/TCP   175m
```

**Comment by matthewmcnew** [comment]
Hi @cmoulliard, I think this might warrant its own issue. 

It should be possible to configure kpack to access a docker registry that is deployed on the same cluster but, the nodes themselves will need to be able to resolve the registry as well.

**Comment by matthewmcnew** [comment]
From the error it looks like there may be a name mismatch between `kind-registry` & `kube-registry`. Could you confirm that is not the case?
